### PR TITLE
409 front door logging

### PR DESCRIPTION
**Pull request by mrzta**
Enable logging for Front Door using the Diagnostics settings 
Create a log analytics workspace 

**Testing** 

- You will require PIM as all the Front Doors are in prod, deploy your own front door or use TSC front door in dev  

- In order to test this set the module source to ```ref=409-front-door-logging```
-  run ```make register domains-infra-apply  ``` (example targets register app)

### CODE REVIEW

**Comment by mrzta** [comment]
[Front door logging](https://trello.com/c/8tCQhHrW/409-front-door-logging)

File: domains/infrastructure/front_door.tf [hunk 1]
```diff
@@ -5,3 +5,29 @@ resource "azurerm_cdn_frontdoor_profile" "main" {
   sku_name            = "Standard_AzureFrontDoor"
   tags                = var.tags
 }
+
+data "azurerm_monitor_diagnostic_categories" "main" {
+  resource_id = values(azurerm_cdn_frontdoor_profile.main)[0].id
+}
+
+resource "azurerm_monitor_diagnostic_setting" "main" {
+  for_each                   = var.hosted_zone
```
Comment (saliceti) [inline domains/infrastructure/front_door.tf:14]: We only want to enable it if monitoring is enabled. Could you amend the condition?

Comment (mrzta) [reply domains/infrastructure/front_door.tf]: amended the condition

File: domains/infrastructure/variables.tf [hunk 2]
```diff
@@ -32,3 +32,8 @@ locals {
     zone_name => merge(zone_cfg, var.deploy_default_records ? local.default_records : null)
   }
 }
+
+variable "azure_enable_monitoring" {
+  type        = bool
+  description = "value to enable monitoring"
```
Comment (saliceti) [inline domains/infrastructure/variables.tf:38]: Maybe change to:
"Enable monitoring and logging in Azure"

File: domains/infrastructure/front_door.tf [hunk 3]
```diff
@@ -5,3 +5,29 @@ resource "azurerm_cdn_frontdoor_profile" "main" {
   sku_name            = "Standard_AzureFrontDoor"
   tags                = var.tags
 }
+
+data "azurerm_monitor_diagnostic_categories" "main" {
+  for_each    = var.azure_enable_monitoring ? var.hosted_zone : {}
+  resource_id = azurerm_cdn_frontdoor_profile.main[each.key].id
```
Comment (saliceti) [inline domains/infrastructure/front_door.tf:11]: Please add blank line above

File: domains/infrastructure/front_door.tf [hunk 4]
```diff
@@ -5,3 +5,29 @@ resource "azurerm_cdn_frontdoor_profile" "main" {
...
+}
+
+resource "azurerm_monitor_diagnostic_setting" "main" {
+  for_each                   = var.azure_enable_monitoring ? var.hosted_zone : {}
+  name                       = "${each.value.front_door_name}-diagnostics"
```
Comment (saliceti) [inline domains/infrastructure/front_door.tf:16]: Please add blank line above

File: domains/infrastructure/front_door.tf [hunk 5]
```diff
@@ -5,3 +5,29 @@ resource "azurerm_cdn_frontdoor_profile" "main" {
...
+  target_resource_id         = azurerm_cdn_frontdoor_profile.main[each.key].id
+  log_analytics_workspace_id = azurerm_log_analytics_workspace.main[each.key].id
+  dynamic "enabled_log" {
```
Comment (saliceti) [inline domains/infrastructure/front_door.tf:19]: Please add blank line above

File: domains/infrastructure/front_door.tf [hunk 6]
```diff
@@ -5,3 +5,29 @@ resource "azurerm_cdn_frontdoor_profile" "main" {
...
+    for_each = data.azurerm_monitor_diagnostic_categories.main[each.key].log_category_types
+    content {
+      category = enabled_log.value
+    }
+  }
+}
+
+resource "azurerm_log_analytics_workspace" "main" {
+  for_each            = var.azure_enable_monitoring ? var.hosted_zone : {}
+  name                = "${each.value.front_door_name}-log"
```
Comment (saliceti) [inline domains/infrastructure/front_door.tf:29]: Please add blank line above

**Review by saliceti — APPROVED** [review]

**Comment by RMcVelia** [comment]
Can we add a doc on using the new logging? It could have some example queries, etc

**Comment by mrzta** [comment]
> Can we add a doc on using the new logging? It could have some example queries, etc

added a doc

**Pull request merged**